openapi3filter/validate_request.go

```diff
@@ -137,7 +138,7 @@ func ValidateParameter(ctx context.Context, input *RequestValidationInput, param
 	}
 
 	// Set default value if needed
-	if value == nil && schema != nil && schema.Default != nil {
+	if !skipSettingDefaults && value == nil && schema != nil && schema.Default != nil {
```

Comment:
Use `input.Options.SkipSettingDefaults` directly :)

Reply:
@fenollp fixed. Please merge if all is OK.